store.go

```diff
@@ -2241,10 +2248,23 @@ func (s *store) DeleteLayer(id string) error {
 		}
 		for _, container := range containers {
 			if container.LayerID == id {
-				return errors.Wrapf(ErrLayerUsedByContainer, "Layer %v used by container %v", id, container.ID)
+				return errors.Wrapf(ErrLayerUsedByContainer, "layer %v used by container %v", id, container.ID)
 			}
 		}
-		return rlstore.Delete(id)
+		if err := rlstore.Delete(id); err != nil {
+			return errors.Wrapf(err, "delete layer %v", id)
+		}
+
+		if istore, ok := ristore.(*imageStore); ok {
```

Comment:
Do we wanna error if `!ok`?

Reply:
it can be `!ok` and that is fine if we don't have to modify `image.MappedTopLayers`.  The check here is just to avoid iterating the images if we don't have too. We already have a check above.  I'll add a comment to make it clearer

Reply:
Thanks!